src/Microsoft.VisualStudio.Editors/Common/Utils.vb

```diff
@@ -314,36 +314,64 @@ Namespace Microsoft.VisualStudio.Editors.Common
                 Else
                     Return Value.ToString()
                 End If
-            Catch ex As Exception
-                RethrowIfUnrecoverable(ex)
+            Catch ex As Exception When Not IsUnrecoverableException(ex)
                 Return "[" & ex.GetType.Name & "]"
             End Try
 #Else
             Return ""
 #End If
         End Function
 
+        Public Function IsUnrecoverableException(ByVal ex As Exception, Optional ByVal ignoreOutOfMemory As Boolean = False) As Boolean
+            Return TypeOf ex Is NullReferenceException _
```

Comment:
Question: Why is NRE considered an unrecoverable exception? Seems very likely that caller can gracefully handle it. Additionally, the definition of the same method in MS.VS.AppDesigner doesn't include NRE (https://github.com/dotnet/roslyn-project-system/blob/master/src/Microsoft.VisualStudio.AppDesigner/Common/Utils.vb#L233)